src/VirtoCommerce.CustomerModule.Web/Module.cs

```diff
@@ -112,14 +113,11 @@ public void PostInitialize(IApplicationBuilder appBuilder)
                 inProcessBus.RegisterHandler<MemberChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<IndexMemberChangedEventHandler>().Handle(message));
             }
 
-            var searchRequestBuilderRegistrar = appBuilder.ApplicationServices.GetService<ISearchRequestBuilderRegistrar>();
-            searchRequestBuilderRegistrar.Register(KnownDocumentTypes.Member, () =>
-            {
-                var searchPhraseParser = appBuilder.ApplicationServices.GetService<ISearchPhraseParser>();
-
-                return new MemberSearchRequestBuilder(searchPhraseParser);
-            });
+            AbstractTypeFactory<MemberSearchRequestBuilder>.RegisterType<MemberSearchRequestBuilder>()
+                .WithFactory(appBuilder.ApplicationServices.GetService<MemberSearchRequestBuilder>);
 
+            var searchRequestBuilderRegistrar = appBuilder.ApplicationServices.GetService<ISearchRequestBuilderRegistrar>();
+            searchRequestBuilderRegistrar.Register(KnownDocumentTypes.Member, AbstractTypeFactory<MemberSearchRequestBuilder>.TryCreateInstance<MemberSearchRequestBuilder>);
```

Comment:
Seems second generic type is redundant AbstractTypeFactory<MemberSearchRequestBuilder>.TryCreateInstance<MemberSearchRequestBuilder>